### ISSUE TITLE
supernova doesn't boot on archlinux

### ISSUE THREAD

**Issue by smoge** [supercollider/supercollider#3251]
In an updated ArchLinux system, with latest git checkout, supernova compiles but can't boot. No problem with scsynth, either with supernova on 3.8 .

```
Class tree inited in 0.01 seconds


*** Welcome to SuperCollider 3.9dev. *** For help type ctrl-c ctrl-h (Emacs) or :SChelp (vim) or ctrl-U (sced/gedit).
sc3> s.boot
-> localhost
sc3> booting server 'localhost' on address: 127.0.0.1:57110
Found 340 LADSPA plugins
JackDriver: client name is 'SuperCollider'
SC_AudioDriver: sample rate = 96000.000000, driver's block size = 1024
JackDriver: connected  system:capture_1 to SuperCollider:in_1
JackDriver: connected  system:capture_2 to SuperCollider:in_2
JackDriver: connected  SuperCollider:out_1 to system:playback_1
JackDriver: connected  SuperCollider:out_2 to system:playback_2
SuperCollider 3 server ready.
JackDriver: max output latency 32.0 ms
Shared memory server interface initialized
Receiving notification messages from server 'localhost'

sc3> s.quitAll
server 'localhost' disconnected shared memory interface
'/quit' sent

-> localhost
sc3> RESULT = 0

sc3> Server.supernova
-> Server
sc3> s.boot
-> localhost
sc3> booting server 'localhost' on address: 127.0.0.1:57110
Supernova booting
Found 340 LADSPA plugins
terminate called without an active exception
RESULT = 0
```

**Comment by smoge** [comment]
```
$ valgrind ./supernova                              (10-23 01:59)
==27358== Memcheck, a memory error detector
==27358== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==27358== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==27358== Command: ./supernova
==27358== 
Supernova booting
compiled for debugging
==27358== Thread 4 Network Receive:
==27358== Syscall param epoll_pwait(sigmask) points to unaddressable byte(s)
==27358==    at 0x67133B6: epoll_pwait (in /usr/lib/libc-2.26.so)
==27358==    by 0x262928: boost::asio::detail::epoll_reactor::run(bool, boost::asio::detail::op_queue<boost::asio::detail::task_io_service_operation>&) (epoll_reactor.ipp:392)
==27358==    by 0x263339: boost::asio::detail::task_io_service::do_run_one(boost::asio::detail::scoped_lock<boost::asio::detail::posix_mutex>&, boost::asio::detail::task_io_service_thread_info&, boost::system::error_code const&) (task_io_service.ipp:356)
==27358==    by 0x2630A4: boost::asio::detail::task_io_service::run(boost::system::error_code&) (task_io_service.ipp:149)
==27358==    by 0x26366E: boost::asio::io_service::run() (io_service.ipp:59)
==27358==    by 0x2638C4: nova::detail::network_thread::start_receive()::{lambda()#1}::operator()() const (osc_server.hpp:52)
==27358==    by 0x26BB22: void std::__invoke_impl<void, nova::detail::network_thread::start_receive()::{lambda()#1}>(std::__invoke_other, nova::detail::network_thread::start_receive()::{lambda()#1}&&) (invoke.h:60)
==27358==    by 0x268837: std::__invoke_result<nova::detail::network_thread::start_receive()::{lambda()#1}>::type std::__invoke<nova::detail::network_thread::start_receive()::{lambda()#1}>(std::__invoke_result&&, (nova::detail::network_thread::start_receive()::{lambda()#1}&&)...) (invoke.h:95)
==27358==    by 0x288565: decltype (__invoke((_S_declval<0ul>)())) std::thread::_Invoker<std::tuple<nova::detail::network_thread::start_receive()::{lambda()#1}> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) (thread:234)
==27358==    by 0x2882BE: std::thread::_Invoker<std::tuple<nova::detail::network_thread::start_receive()::{lambda()#1}> >::operator()() (thread:243)
==27358==    by 0x287F55: std::thread::_State_impl<std::thread::_Invoker<std::tuple<nova::detail::network_thread::start_receive()::{lambda()#1}> > >::_M_run() (thread:186)
==27358==    by 0x5DECA6E: execute_native_thread_routine (thread.cc:83)
==27358==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==27358== 
Found 340 LADSPA plugins
terminate called without an active exception
==27358== 
==27358== Process terminating with default action of signal 6 (SIGABRT): dumping core
==27358==    at 0x66518A0: raise (in /usr/lib/libc-2.26.so)
==27358==    by 0x6652F08: abort (in /usr/lib/libc-2.26.so)
==27358==    by 0x5DC1D76: __gnu_cxx::__verbose_terminate_handler() (vterminate.cc:95)
==27358==    by 0x5DBF8E5: __cxxabiv1::__terminate(void (*)()) (eh_terminate.cc:47)
==27358==    by 0x5DBF932: std::terminate() (eh_terminate.cc:57)
==27358==    by 0x11B565FC: load (in /usr/lib/SuperCollider/plugins/UIUGens_supernova.so)
==27358==    by 0x24A823: nova::sc_ugen_factory::load_plugin(boost::filesystem::path const&) (sc_ugen_factory.cpp:292)
==27358==    by 0x24A465: nova::sc_ugen_factory::load_plugin_folder(boost::filesystem::path const&) (sc_ugen_factory.cpp:234)
==27358==    by 0x1900D2: (anonymous namespace)::set_plugin_paths(nova::server_arguments const&, nova::sc_ugen_factory*) (main.cpp:254)
==27358==    by 0x190954: main (main.cpp:383)
==27358== 
==27358== HEAP SUMMARY:
==27358==     in use at exit: 5,181,672 bytes in 13,131 blocks
==27358==   total heap usage: 271,410 allocs, 258,279 frees, 50,751,442 bytes allocated
==27358== 
==27358== LEAK SUMMARY:
==27358==    definitely lost: 176,910 bytes in 2,499 blocks
==27358==    indirectly lost: 16,384 bytes in 1 blocks
==27358==      possibly lost: 207,269 bytes in 112 blocks
==27358==    still reachable: 4,768,901 bytes in 10,422 blocks
==27358==                       of which reachable via heuristic:
==27358==                         length64           : 1,688 bytes in 32 blocks
==27358==                         newarray           : 198,472 bytes in 38 blocks
==27358==         suppressed: 0 bytes in 0 blocks
==27358== Rerun with --leak-check=full to see details of leaked memory
==27358== 
==27358== For counts of detected and suppressed errors, rerun with: -v
==27358== ERROR SUMMARY: 2 errors from 1 contexts (suppressed: 0 from 0)
[1]    27358 abort (core dumped)  valgrind ./supernova
```

**Comment by smoge** [comment]
```

$ gdb ./supernova
GNU gdb (GDB) 8.0.1
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./supernova...done.
(gdb) r
Starting program: /home/smoge/Builds/srcpackages/supercollider-git/pkg/supercollider-git/usr/bin/supernova 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Supernova booting
compiled for debugging
[New Thread 0x7ffff4aa3700 (LWP 28758)]
[New Thread 0x7ffff3a21700 (LWP 28760)]
[New Thread 0x7ffff3220700 (LWP 28761)]
[New Thread 0x7fffed473700 (LWP 28762)]
[New Thread 0x7fffd60fd700 (LWP 28765)]
[New Thread 0x7fffd58fc700 (LWP 28766)]
Found 340 LADSPA plugins
[New Thread 0x7fffbf73e700 (LWP 28768)]
[New Thread 0x7fffbdc44700 (LWP 28770)]
terminate called without an active exception

Thread 1 "supernova" received signal SIGABRT, Aborted.
0x00007ffff62718a0 in raise () from /usr/lib/libc.so.6
(gdb) bt full
#0  0x00007ffff62718a0 in raise () at /usr/lib/libc.so.6
#1  0x00007ffff6272f09 in abort () at /usr/lib/libc.so.6
#2  0x00007ffff6be5d77 in __gnu_cxx::__verbose_terminate_handler() () at /build/gcc/src/gcc/libstdc++-v3/libsupc++/vterminate.cc:95
        terminating = true
        t = <optimized out>
#3  0x00007ffff6be38e6 in __cxxabiv1::__terminate(void (*)()) (handler=<optimized out>)
    at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:47
#4  0x00007ffff6be3933 in std::terminate() () at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:57
#5  0x00007fffede395fd in load () at /usr/lib/SuperCollider/plugins/UIUGens_supernova.so
#6  0x0000000100142824 in nova::sc_ugen_factory::load_plugin(boost::filesystem::path const&) (this=0x10065d020, path=...)
    at /home/smoge/Builds/srcpackages/supercollider-git/src/supercollider/server/supernova/sc/sc_ugen_factory.cpp:292
        handle = 0x100704ef0
        api_version = 0x7fffede39470 <api_version>
        supernova_check = 0x7fffede39480 <server_type>
        load_symbol = 0x7fffede39490 <load>
        load_func = 0x7fffede39490 <load>
#7  0x0000000100142466 in nova::sc_ugen_factory::load_plugin_folder(boost::filesystem::path const&) (this=0x10065d020, path=...)
    at /home/smoge/Builds/srcpackages/supercollider-git/src/supercollider/server/supernova/sc/sc_ugen_factory.cpp:234
        it = 
              {<boost::iterators::iterator_facade<boost::filesystem::directory_iterator, boost::filesystem::directory_entry, boost::iterators::single_pass_traversal_tag, boost::filesystem::directory_entry&, long>> = {<boost::iterators::detail::iterator_facade_base<boost::filesystem::directory_iterator, boost::filesystem::directory_entry, boost::iterators::single_pass_traversal_tag, boost::filesystem::directory_entry&, long, false, false>> = {<No data fields>}, <No data fields>}, m_imp = {px = 0x100816530, pn = {pi_ = 0x1008164a0}}}
        end = 
              {<boost::iterators::iterator_facade<boost::filesystem::directory_iterator, boost::filesystem::directory_entry, boost::iterators::single_pass_traversal_tag, boost::filesystem::directory_entry&, long>> = {<boost::iterators::detail::iterator_facade_base<boost::filesystem::directory_iterator, boost::filesystem::directory_entry, boost::iterators::single_pass_traversal_tag, boost::filesystem::directory_entry&, long, false, false>> = {<No data fields>}, <No data fields>}, m_imp = {px = 0x0, pn = {pi_ = 0x0}}}
#8  0x00000001000880d3 in (anonymous namespace)::set_plugin_paths(nova::server_arguments const&, nova::sc_ugen_factory*) (args=..., factory=0x10065d020) at /home/smoge/Builds/srcpackages/supercollider-git/src/supercollider/server/supernova/server/main.cpp:254
        folder = 
          @0x100564120: {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib64/SuperCollider/plugins"}
        __for_range = 
            std::vector of length 5, capacity 5 = {{static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/local/lib/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib64/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge/.local/share/SuperCollider/Extensions"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge/share/SuperCollider/plugins"}}
        __for_begin = 
  {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib64/SuperCollider/plugins"}
        __for_end = 
---Type <return> to continue, or q <return> to quit---
  {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = <error reading variable: Cannot create a lazy string with address 0x0, and a non-zero length.>}
        home = {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge"}
        folders = 
            std::vector of length 5, capacity 5 = {{static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/local/lib/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib64/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge/.local/share/SuperCollider/Extensions"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge/share/SuperCollider/plugins"}}
#9  0x0000000100088955 in main(int, char**) (argc=1, argv=0x7fffffffe2a8)
    at /home/smoge/Builds/srcpackages/supercollider-git/src/supercollider/server/supernova/server/main.cpp:383
        args = 
                @0x100570080: {udp_port = 0, tcp_port = 0, control_busses = 16384, audio_busses = 1024, blocksize = 64, samplerate = 44100, hardware_buffer_size = 0, buffers = 1024, max_nodes = 1024, max_synthdefs = 1024, use_system_clock = 0, rt_pool_size = 8192, wires = 64, rng_count = 64, load_synthdefs = 1, verbosity = 0, dump_version = false, hw_name = std::vector of length 0, capacity 0, memory_locking = false, threads = 4, ugen_paths = std::vector of length 0, capacity 0, restrict_paths = std::vector of length 0, capacity 0, input_channels = 8, output_channels = 8, server_password = "", non_rt = false, command_file = "", input_file = "", output_file = "", header_format = "", sample_format = "", static instance_ = {_M_t = {_M_t = std::tuple containing = {[1] = 0x100570080, [2] = {<std::default_delete<nova::server_arguments>> = {<No data fields>}, <No data fields>}}}}}
        server = Python Exception <class 'gdb.error'> value of type `nova::nova_server' requires 874056 bytes, which is more than max-value-size: 
(gdb) q
A debugging session is active.

	Inferior 1 [process 28754] will be killed.

Quit anyway? (y or n) y
```

**Comment by smoge** [comment]
I recompiled sc3-plugins just before that, it's not mismatched API, I think

**Comment by smoge** [comment]
I don't think is api mismatch, I recompiled sc3-plugins and it works on scsynth

**Comment by smoge** [comment]
probably I'm just missing something obvious

**Comment by brianlheim** [comment]
that's a segfault all right...  the fact that it is crashing in the load_plugin routine is suspicious. I hope i didn't accidentally do anything to cause that. @smoge can you try removing `UIUGens_supernova.so` from your library paths and see if you get this same issue? If so it's probably my code.

**Comment by brianlheim** [comment]
Hm, actually, it looks like this might be a dupe of this issue: https://github.com/supercollider/supercollider/issues/968

Trace looks pretty much the same.

**Comment by smoge** [comment]
Same thing

'''
$ gdb ./supernova        
GNU gdb (GDB) 8.0.1
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./supernova...done.
(gdb) r
Starting program: /home/smoge/Builds/srcpackages/supercollider-git/pkg/supercollider-git/usr/bin/supernova 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Supernova booting
compiled for debugging
[New Thread 0x7ffff4aa3700 (LWP 26595)]
[New Thread 0x7ffff3a21700 (LWP 26596)]
[New Thread 0x7ffff3220700 (LWP 26597)]
[New Thread 0x7fffeb1a7700 (LWP 26598)]
[New Thread 0x7fffd6039700 (LWP 26599)]
[New Thread 0x7fffd5838700 (LWP 26600)]
Found 340 LADSPA plugins
[New Thread 0x7fffbf53b700 (LWP 26601)]
[New Thread 0x7fffbda41700 (LWP 26602)]
terminate called without an active exception

Thread 1 "supernova" received signal SIGABRT, Aborted.
0x00007ffff62718a0 in raise () from /usr/lib/libc.so.6
(gdb) bt full
#0  0x00007ffff62718a0 in raise () at /usr/lib/libc.so.6
#1  0x00007ffff6272f09 in abort () at /usr/lib/libc.so.6
#2  0x00007ffff6be5d77 in __gnu_cxx::__verbose_terminate_handler() ()
    at /build/gcc/src/gcc/libstdc++-v3/libsupc++/vterminate.cc:95
        terminating = true
        t = <optimized out>
#3  0x00007ffff6be38e6 in __cxxabiv1::__terminate(void (*)()) (handler=<optimized out>) at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:47
#4  0x00007ffff6be3933 in std::terminate() ()
    at /build/gcc/src/gcc/libstdc++-v3/libsupc++/eh_terminate.cc:57
#5  0x00007fffebb6e5fd in load ()
    at /usr/lib/SuperCollider/plugins/UIUGens_supernova.so.bak
#6  0x0000000100142824 in nova::sc_ugen_factory::load_plugin(boost::filesystem::path const&) (this=0x10065d020, path=...)
    at /home/smoge/Builds/srcpackages/supercollider-git/src/supercollider/server/supernova/sc/sc_ugen_factory.cpp:292
        handle = 0x100713cd0
        api_version = 0x7fffebb6e470 <api_version>
        supernova_check = 0x7fffebb6e480 <server_type>
        load_symbol = 0x7fffebb6e490 <load>
        load_func = 0x7fffebb6e490 <load>
#7  0x0000000100142466 in nova::sc_ugen_factory::load_plugin_folder(boost::filesystem::path const&) (this=0x10065d020, path=...)
---Type <return> to continue, or q <return> to quit---
    at /home/smoge/Builds/srcpackages/supercollider-git/src/supercollider/server/supernova/sc/sc_ugen_factory.cpp:234
        it = 
              {<boost::iterators::iterator_facade<boost::filesystem::directory_iterator, boost::filesystem::directory_entry, boost::iterators::single_pass_traversal_tag, boost::filesystem::directory_entry&, long>> = {<boost::iterators::detail::iterator_facade_base<boost::filesystem::directory_iterator, boost::filesystem::directory_entry, boost::iterators::single_pass_traversal_tag, boost::filesystem::directory_entry&, long, false, false>> = {<No data fields>}, <No data fields>}, m_imp = {px = 0x1008166b0, pn = {pi_ = 0x100590f40}}}
        end = 
              {<boost::iterators::iterator_facade<boost::filesystem::directory_iterator, boost::filesystem::directory_entry, boost::iterators::single_pass_traversal_tag, boost::filesystem::directory_entry&, long>> = {<boost::iterators::detail::iterator_facade_base<boost::filesystem::directory_iterator, boost::filesystem::directory_entry, boost::iterators::single_pass_traversal_tag, boost::filesystem::directory_entry&, long, false, false>> = {<No data fields>}, <No data fields>}, m_imp = {px = 0x0, pn = {pi_ = 0x0}}}
#8  0x00000001000880d3 in (anonymous namespace)::set_plugin_paths(nova::server_arguments const&, nova::sc_ugen_factory*) (args=..., factory=0x10065d020)
    at /home/smoge/Builds/srcpackages/supercollider-git/src/supercollider/server/supernova/server/main.cpp:254
        folder = 
---Type <return> to continue, or q <return> to quit---
          @0x100564120: {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib64/SuperCollider/plugins"}
        __for_range = 
            std::vector of length 5, capacity 5 = {{static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/local/lib/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib64/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge/.local/share/SuperCollider/Extensions"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge/share/SuperCollider/plugins"}}
        __for_begin = 
  {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib64/SuperCollider/plugins"}
        __for_end = 
  {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = <error reading variable: Cannot create a lazy string with address 0x0, and a non-zero length.>}
        home = 
          {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge"}
---Type <return> to continue, or q <return> to quit---
        folders = 
            std::vector of length 5, capacity 5 = {{static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/local/lib/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/usr/lib64/SuperCollider/plugins"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge/.local/share/SuperCollider/Extensions"}, {static separator = 47 '/', static preferred_separator = 47 '/', static dot = 46 '.', m_pathname = "/home/smoge/share/SuperCollider/plugins"}}
#9  0x0000000100088955 in main(int, char**) (argc=1, argv=0x7fffffffe238)
    at /home/smoge/Builds/srcpackages/supercollider-git/src/supercollider/server/supernova/server/main.cpp:383
        args = 
                @0x100570080: {udp_port = 0, tcp_port = 0, control_busses = 16384, audio_busses = 1024, blocksize = 64, samplerate = 44100, hardware_buffer_size = 0, buffers = 1024, max_nodes = 1024, max_synthdefs = 1024, use_system_clock = 0, rt_pool_size = 8192, wires = 64, rng_count = 64, load_synthdefs = 1, verbosity = 0, dump_version = false, hw_name = std::vector of length 0, capacity 0, memory_locking = false, threads = 4, ugen_paths = std::vector of length 0, capacity 0, restrict_paths = std::vector of length 0, capacity 0, input_channels = 8, output_channels = 8, server_password = "", non_rt = false, command_file = "", input_---Type <return> to continue, or q <return> to quit---
file = "", output_file = "", header_format = "", sample_format = "", static instance_ = {_M_t = {_M_t = std::tuple containing = {[1] = 0x100570080, [2] = {<std::default_delete<nova::server_arguments>> = {<No data fields>}, <No data fields>}}}}}
        server = Python Exception <class 'gdb.error'> value of type `nova::nova_server' requires 874056 bytes, which is more than max-value-size: 
(gdb) 
'''

**Comment by smoge** [comment]
I can try a `git bisect `later until a find the commit, when I have the time

**Comment by brianlheim** [comment]
@smoge this time it's loading `UIUGens_supernova.so.bak`, can you remove that too?

**Comment by smoge** [comment]
Yes, I remove this one and the one in /usr/lib, now supernova boots normally

**Comment by brianlheim** [comment]
ok great, that's a huge help!

**Comment by smoge** [comment]
any news on that?

**Comment by brianlheim** [comment]
no, sorry - i just got back to school and swamped with work now.

**Comment by brianlheim** [comment]
@smoge - have you had a chance to bisect it? i probably won't have time in the next few weeks, school has gotten really busy as november break approaches.

**Comment by smoge** [comment]
On 11/2/17 18:35, Brian Heim wrote:
> @smoge <https://github.com/smoge> - have you had a chance to bisect it? 
> i probably won't have time in the next few weeks, school has gotten 
> really busy as november break approaches.

I understood you got it. I don't have that much time in my hands either, 
but I will try!

**Comment by capocasa** [comment]
I tried to find the commit that introduced the problem.

The earliest I managed to build (with GCC7) was 3.7.0. That still has the issue for me.

My gdb stacktrace only mentions libjack and libpthread, but not supernova itself.

I also noticed that when using jack2, the problem doesn't happen, so at least there's a temporary workaround.

So I suspect that the change that triggered the issue happened in jack 1, but that's as far as I managed to push right now.
